### PR TITLE
Admin UI Identity Quotas

### PR DESCRIPTION
**Pull request by JLSRKonk**
# Readiness checklist

- [ ] I added/updated unit tests.
- [ ] I added/updated integration tests.
- [x] I ensured that the PR title is good enough for the changelog.
- [x] I labeled the PR.

### CODE REVIEW

**Comment by JLSRKonk** [comment]
Screenshots:
identity

identity-quota

**Review by tnotheis — CHANGES_REQUESTED** [review]
I would suggest getting rid of the "DTO" postfix. The Backbone only needs it to distinguish it from the domain classes.

File: AdminUI/src/app/services/quotas-service/quotas.service.ts [hunk 1]
```diff
@@ -25,36 +25,53 @@ export class QuotasService {
...
+
+export interface TierQuotaDefinitionDTO {
+    id: string;
+    metric: MetricDTO;
     max: number;
     period: string;
 }
 
-export interface Metric {
+export interface IdentityQuotaDefinitionDTO {
```
[the remark this answers is not in the document]
Comment (JLSRKonk) [reply AdminUI/src/app/services/quotas-service/quotas.service.ts:60]: We don't have quotas for Identities in the main branch, as far as I am aware. I got this DTO from the open pull request "Quotas For Identities: application #175". We have a DTO definition with this name there. https://github.com/nmshd/backbone/pull/175/files#diff-8a89fce308c5fb982cdce41f36360db55512b5237a4eceaaa83ced4dae39a205

File: AdminUI/src/app/components/quotas/identity/identity-edit/identity-edit.component.html [hunk 2]
```diff
@@ -5,37 +5,26 @@ <h2>{{ header }}</h2>
...
+                    <span matListItemLine>{{ identity.clientId }}</span>
+                </mat-list-item>
+                <mat-list-item>
+                    <span matListItemTitle>Public Key</span>
+                    <span matListItemLine>{{ identity.publicKey }}</span>
+                </mat-list-item>
+                <mat-list-item>
+                    <span matListItemTitle>Created At</span>
+                    <span matListItemLine>{{
+                        dateConvert(identity.createdAt)
```
[the remark this answers is not in the document]
Comment (JLSRKonk) [reply AdminUI/src/app/components/quotas/identity/identity-edit/identity-edit.component.html:24]: When we use a date variable like this, without any kind of processing, the full date date string will be displayed. I can see two ways to solve this.

1. Use angular DatePipes. In this scenario, we would use the DatePipe like this `{{ identity.createdAt | date }}` and the date would be properly displayed. We can customize the way the DatePipe displays dates globally.
Here is the documentation: https://angular.io/api/common/DatePipe , https://angular.io/api/common/DATE_PIPE_DEFAULT_OPTIONS

2. Alternatively, we can manipulate the date in the data model, to change its type, and display it directly.

[the remark this answers is not in the document]
Comment (tnotheis) [reply AdminUI/src/app/components/quotas/identity/identity-edit/identity-edit.component.html:24]: The latter is the solution I would have chosen for an MVVM application. But I don't know how to do the MVC way.

File: AdminUI/src/app/services/quotas-service/quotas.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tnotheis) [reply AdminUI/src/app/services/quotas-service/quotas.service.ts:60]: Ohh, okay, then it is incorrect there as well. :) It should be called IdentityQuotaDTO - or, if we get rid of the DTO prefix - just IdentityQuota.

File: AdminUI/src/app/app.module.ts [hunk 3]
```diff
@@ -79,7 +80,13 @@ import { AssignQuotasDialogComponent } from './components/quotas/assign-quotas-d
         MatSelectModule,
         MatChipsModule,
     ],
-    providers: [SidebarService],
+    providers: [
+        SidebarService,
+        {
+            provide: DATE_PIPE_DEFAULT_OPTIONS,
+            useValue: { dateFormat: 'd/M/yyyy' },
```
Comment (tnotheis) [inline AdminUI/src/app/app.module.ts:87]: As the majority of the users will probably be German, can you use `dd.mm.yyyy` as the date format please? Plus: the time would be helpful as well, so you should change it to `dd.mm.yyyy HH:mm:ss`

**Review by tnotheis — CHANGES_REQUESTED** [review]

**Review by tnotheis — APPROVED** [review]

**Review by tnotheis — CHANGES_REQUESTED** [review]
Wait

**Comment by tnotheis** [comment]
Don't merge this to main. It should go into the feature branch for identity quota

**Review by tnotheis — APPROVED** [review]

**Pull request merged**